pkg/storage/backend-storage/backend-storage.go

```diff
@@ -71,24 +74,125 @@ func IsBackendStorageNeededForVM(vm *corev1.VirtualMachine) bool {
 	return HasPersistentTPMDevice(&vm.Spec.Template.Spec)
 }
 
-func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) error {
+type BackendStorage struct {
```

Comment:
I wonder if there is a better name for this type of storage (TPM/EFI) "backend" is pretty generic.  I could imagine a new developer to the codebase finding this and being confused.  Maybe SystemStorage?  I dunno just throwing it out there

Reply:
I know I hate that name! (yes I came up with it :sweat_smile:)
SystemStorage is better, although a bit generic as well...
EmulatorStorage? I don't know, naming is hard...

Reply:
I like `EmulatorStorage` the best so far...

Reply:
Do we want to make any changes on naming?

Reply:
That would definitely be a separate PR in my opinion.
It's also probably something we'd want to discuss with a wider audience, as that name has been around since the design proposal, as bad as it is...
I honestly didn't think it was that bad until I realized, much later, that the term "backend storage" was already occasionally used to designate the storage medium used for PVCs.
Feel free to open an issue or send an email to kubevirt-dev about this, or I'm happy to take care of it if you think that makes sense.

Reply:
Since this is a brand new struct I didn't think it would be such a big deal

Reply:
Oh sure, but the struct is named after the package, which is named after the feature!
If we don't rename everything, using a different name for the struct would just be confusing...